driver-async/src/main/com/mongodb/async/client/FetchAllObserver.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client;
+
+class FetchAllObserver<TResult> implements Observer<TResult> {
+
+    private final Observer<TResult> delegate;
+    private final long batchSize;
+    private long counter = 0;
+    private Subscription subscription;
+
+    public FetchAllObserver(final Observer<TResult> delegate, final Integer batchSize) {
```

Comment:
It's not clear to me why there needs to be a batch size here.  The batch size is already set on the MongoIterable, which will apply it to the the underlying cursor.  Given that, can't this class simply request Long.MAX_VALUE and rely on the delegate to apply any batch size?